mount/mount.go

```diff
@@ -22,3 +46,39 @@ func All(mounts []Mount, target string) error {
 	}
 	return nil
 }
+
+// TempMount mounts the provided mounts to a temp dir, and pass the temp dir to f.
+// Finally we will unmount and remove the temp dir regardless of the result of f.
+func TempMount(mounts []Mount, m Mounter, f func(root string) error) error {
+	if len(mounts) == 0 {
+		return errdefs.ErrInvalidArgument
+	}
+	if m == nil {
+		m = mounter{}
+	}
+
+	root, err := ioutil.TempDir("", "containerd-TempMount")
+	if err != nil {
+		return errors.Wrapf(err, "failed to create %s", root)
+	}
+	// We change RemoveAll to Remove so that we either leak a temp dir
+	// if it fails but not RM snapshot data. refer to #1868 #1785
+	defer os.Remove(root)
```

Comment:
Check the error in a closure and log the error.

Reply:
done